src/main/java/eu/bbmri_eric/negotiator/service/MolgenisService.java

```diff
@@ -27,4 +28,19 @@ public interface MolgenisService {
    * @return Optional<MolgenisBiobank>
    */
   Optional<MolgenisBiobank> findBiobankById(String id);
+
+  /**
+   * Fetch all BBMRI-ERIC, MIABIS defined Biobanks.
+   *
+   * @return Optional List<MolgenisBiobank>
+   */
+  List<MolgenisBiobank> findAllBiobanks();
```

Comment:
no need to use the MolgenisService, tbh I would delete it. Just use the discovery service client